pkg/plugins/golang/v3/scaffolds/internal/templates/main.go

```diff
@@ -213,13 +214,20 @@ func init() {
 }
 
 func main() {
+{{- if not .ComponentConfig }}
 	var metricsAddr string
 	var enableLeaderElection bool
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. " +
 		"Enabling this will ensure there is only one active controller manager.")
-
+{{- else }}
+  var configFile string
+	flag.StringVar(&configFile, "config", "controller_manager_config.yaml", 
+		"The controller will load its initial configuration from this file. " +
+		"Omit this flag to use the default configuration values. " +
+		"Command-line flags override configuration from this file.")
+{{ end }}
```

Comment:
Nit: preserve previous spacing
```suggestion
{{- end }}

```